packages/frontend/src/lib/disk-image/DiskImageDetailsVirtualMachine.svelte

```diff
@@ -0,0 +1,309 @@
+
+  // Initialize the terminal with the correct theme.
+  // we set disableStdin to true since we are using the attach addon which causes conflict if
+  // we try to use stdin.
+  logsTerminal = new Terminal({
+    fontSize: fontSize,
+    lineHeight: lineHeight,
+    theme: getTerminalTheme(),
+    disableStdin: true,
+  });
+
+  // Wait for the port to become available before trying to connect to the socket, as there is
+  // no point in trying the connection until the port is available.
+  // exit early if so.
+  socketStatus = 'Waiting for WebSocket connection to become available...';
+  const portAvailable = await waitForPort(45252);
+  if (!portAvailable) {
+    console.error('Port 45252 is not available, exiting...');
+    return;
+  }
+
+  // Try to connect to the socket and error if we are unable to / end early.
+  try {
+    socket = new WebSocket(`ws://127.0.0.1:45252`);
+    socket.binaryType = 'arraybuffer';
+
+    // Bind the event handlers
+    socket.onclose = closeHandler;
+    socket.onopen = openHandler;
+    socket.onerror = errorHandler;
+    socketStatus = 'WebSocket connection established, waiting for VM to boot...';
+  } catch (error) {
+    vmLaunchError = 'Failed to connect to the socket';
+    console.error('Failed to connect to the socket:', error);
+    return;
+  }
+
+  // Addons for the terminal (fit and attach)
+  termFit = new FitAddon();
+  logsTerminal.loadAddon(termFit);
+  logsTerminal.open(logsXtermDiv);
+
+  // Call fit addon each time we resize the window
+  window.addEventListener('resize', () => {
+    termFit.fit();
+  });
+  termFit.fit();
+
+  // Addon for attaching to the socket
+  // we load this LAST so that the resizer works correctly (as GRUB output is not resized correctly on first boot)
+  // to avoid any conflicts with the resizer.
+  attachAddon = new AttachAddon(socket);
+  logsTerminal.loadAddon(attachAddon);
+
+  // This is required for keyboard input to work since we are using the attach addon
+  logsTerminal.onKey((e: any) => {
+    e = e.key;
+    if (socket !== undefined) {
+      const encoder = new TextEncoder();
+      const binaryData = encoder.encode(e);
+      socket.send(binaryData.buffer);
+    }
+  });
+
+  // Allow copy / paste support with Ctrl+V
+  logsTerminal.attachCustomKeyEventHandler((arg: KeyboardEvent) => {
+    // Allow copy and paste from the ctrl key (windows and linux)
+    // and "meta" key (macOS). Sometimes on macOS the ctrl key is used, so we safely assume that either meta or ctrl is meant
+    // for paste.
+    if ((arg.ctrlKey || arg.metaKey) && arg.code === 'KeyV' && arg.type === 'keydown') {
+      bootcClient.readFromClipboard().then(clipboardText => {
+        // Send to socket
+        if (socket !== undefined) {
+          const encoder = new TextEncoder();
+          const binaryData = encoder.encode(clipboardText);
+          socket.send(binaryData.buffer);
+        }
+      });
+    }
+    return true;
+  });
+}
```

Comment:
why do we need that, I can copy/paste in Terminal of Podman Desktop without this

Reply:
The reason is because we are using the special websockets mode for addon-attach https://www.npmjs.com/package/xterm-addon-attach and unfortunately we have to "disable stdin" since it's more meant for viewing websocket connections rather than interaction.

That's why we have to add a custom event handler for copying and pasting code as well as interpret each keystroke (see key section code).